RecoEgamma/EgammaTools/interface/Spot.h

```diff
@@ -0,0 +1,37 @@
+#ifndef SPOT_H
+#define SPOT_H
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/ForwardDetId/interface/ForwardSubdetector.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
```

Comment:
why is the ForwardSubdetector include needed here?
The Spot class implementation looks rather non-specific [ignoring apparently not necessary dependence on HGCalDetId].
If this can work only for HGC, I think all of the new code better reflect it in the name.